### PR TITLE
eslint: enable valid-jsdoc

### PR DESCRIPTION
**Pull request by foxtree**
The `valid-jsdoc` rule is built-in to ESLint:  https://eslint.org/docs/rules/valid-jsdoc

For obvious reasons, this is not enabled by default, via the ESLint `airbnb` suite that we are using.
(e.g.: consumers are not necessarily using jsdocs)

image

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ class Client {
   /**
    * Get the httpClient's baseURL value.
    *
-   * @return {String}
+   * @return {String} - Get the httpClient baseURL
```
Comment (bkaney) [inline lib/client.js]: since httpClient is internal implementation / private, I think the documentation should mention it.

Comment (Jammjammjamm) [reply lib/client.js]: Should we remove all of the httpClient and ReferenceResolver documentation?

Comment (bkaney) [reply lib/client.js]: we did :)

Comment (foxtree) [reply lib/client.js]: fixed.

File: lib/client.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ class Client {
   /**
    * Get the httpClient's baseURL value.
```
Comment (bkaney) [inline lib/client.js]: This should also not reference `httpClient`

File: lib/client.js [hunk 3]
```diff
@@ -29,7 +29,7 @@ class Client {
   /**
    * Set the httpClient's baseURL value to a URL string.
```
Comment (bkaney) [inline lib/client.js]: ditto

**Comment by foxtree** [comment]
Note that commit `d9688` removes generation of docs for private members: 

This PR deletes the `docs` directory, before regenerating.

**Comment by foxtree** [comment]
https://eslint.org/docs/rules/valid-jsdoc#prefertype


```
"valid-jsdoc": ["error", { "preferType": { "Boolean": "boolean", "Number": "number", "object": "Object", "String": "string" } }]
```

**Pull request merged**